django/db/models/query.py

```diff
@@ -1513,23 +1513,6 @@ def _filter_or_exclude_inplace(self, negate, args, kwargs):
         else:
             self._query.add_q(Q(*args, **kwargs))
 
-    def complex_filter(self, filter_obj):
```

Comment:
Should we leave a deprecation shim around? This function has been around forever (it predated support for `Q` IIRC) so even if it's not part of the public API we should consider adding a shim.